test/unit/server/simulate_transaction_test.js

```diff
@@ -136,34 +87,60 @@ describe('Server#simulateTransaction', function () {
   });
 
   it('works when there are no results', function () {
-    const simResponseCopy = JSON.parse(JSON.stringify(simulationResponse));
-    delete simResponseCopy.results;
-
+    const simResponse = baseSimulationResponse();
     const parsedCopy = cloneSimulation(parsedSimulationResponse);
     delete parsedCopy.result;
 
-    const parsed = SorobanClient.parseRawSimulation(simResponseCopy);
+    const parsed = SorobanClient.parseRawSimulation(simResponse);
     expect(parsed).to.deep.equal(parsedCopy);
   });
 
   it('works with no auth', function () {
-    const simResponseCopy = JSON.parse(JSON.stringify(simulationResponse));
-    delete simResponseCopy.results[0].auth;
+    const simResponse = invokeSimulationResponse(address);
+    delete simResponse.results[0].auth;
 
     const parsedCopy = cloneSimulation(parsedSimulationResponse);
     parsedCopy.result.auth = [];
-    const parsed = SorobanClient.parseRawSimulation(simResponseCopy);
+    const parsed = SorobanClient.parseRawSimulation(simResponse);
 
     // FIXME: This is a workaround for an xdrgen bug that does not allow you to
-    // build "perfectly-equal" xdr.ExtensionPoint instances (but they will still
-    // be binary-equal, so the test passes), but it should be fixed once we
-    // upgrade the XDR to the final testnet version.
+    // build "perfectly-equal" xdr.ExtensionPoint instances (but they're still
+    // binary-equal, so the test passes).
     parsedCopy.transactionData = parsedCopy.transactionData.build();
     parsed.transactionData = parsed.transactionData.build();
 
     expect(parsed).to.be.deep.equal(parsedCopy);
```

Comment:
these tests that parse raw into type specific sim response, also seem like a good place to add coverage on those new helper functions:

```
expect(IsSimulationSuccess(parsed)).to.equal(true);

```